### PR TITLE
Implementing fragments

### PR DESCRIPTION
**Pull request by daniel-waiguru**
Added a RecyclerView in  ``` MovieFragement.kt ``` to display a list of movies

### CODE REVIEW

**Comment by abunur** [comment]
Hi Daniel, you should name the PR appropriately. "Implementing fragments" is not descriptive of this week's homework assignment. For future homework assignments please use the naming convention feature/week5-hw et.

Good job completing the assignment. Your layout needs some improvement. In landscape view, the banner image takes up most of the screen space and there is not much room to see the recycler view. The images in the recycler view in landscape view become warped. You should try to maintain the image aspect ratio.

Your grade for this assignment is 'good job'

File: Week04/MoviesApp/app/src/main/java/tech/danielwaiguru/moviesapp/MovieFragment.kt [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import androidx.recyclerview.widget.GridLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import kotlinx.android.synthetic.main.fragment_movie.view.*
+
+
+class MovieFragment : Fragment() {
+    private lateinit var movieRecyclerView: RecyclerView
+    private lateinit var movieAdapter: MovieAdapter
+
+    override fun onCreate(savedInstanceState: Bundle?) {
```
Comment (abunur) [inline Week04/MoviesApp/app/src/main/java/tech/danielwaiguru/moviesapp/MovieFragment.kt]: Pay attention to the warnings from the IDE. There is no reason to include an override if you are not doing anything other than calling super.

File: Week04/MoviesApp/app/src/main/res/layout/activity_main.xml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".MainActivity">
+
+    <com.google.android.material.appbar.AppBarLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content">
+
+    </com.google.android.material.appbar.AppBarLayout>
+    <fragment
```
Comment (abunur) [inline Week04/MoviesApp/app/src/main/res/layout/activity_main.xml]: pay attention to warnings from the IDE, it also shows you how to fix them.
this should be

androidx.fragment.app.FragmentContainerView

File: Week04/MoviesApp/app/src/main/res/layout/movie_item_view_holder.xml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    app:cardCornerRadius="6dp"
+    android:elevation="10dp"
+    android:layout_margin="8dp"
+    xmlns:app="http://schemas.android.com/apk/res-auto">
+    <androidx.constraintlayout.widget.ConstraintLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content">
+        <ImageView
```
Comment (abunur) [inline Week04/MoviesApp/app/src/main/res/layout/movie_item_view_holder.xml:12]: make sure you include contentDescription for all images, for accessibility as we discussed previously.

File: Week04/MoviesApp/app/src/main/res/layout/movie_item_view_holder.xml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+            android:id="@+id/posterImageView"
+            android:layout_width="wrap_content"
+            android:layout_height="200dp"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent"
+            app:layout_constraintBottom_toBottomOf="parent"
+            android:src="@drawable/crossings"
+            android:scaleType="fitXY"
+            />
+        <!--<TextView
```
Comment (abunur) [inline Week04/MoviesApp/app/src/main/res/layout/movie_item_view_holder.xml]: Do not leave commented out code - this should be deleted. If you ever want to go back, that's what git history is for.

File: Week04/MoviesApp/app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    <string name="app_name">MoviesApp</string>
+    <string name="about">About</string>
+    <string name="app_information">App information</string>
+    <string name="share">Share</string>
+    <string name="share_the_app_with_friends_and_family">Share the app with friends and family</string>
+    <string name="appsettings">App\'s settings</string>
+    <string name="settings">Settings</string>
+    <string name="search_your_favorite_movie">Search your favorite movie</string>
+    <string name="search_title">Search</string>
+    <!-- TODO: Remove or change this placeholder text -->
```
Comment (abunur) [inline Week04/MoviesApp/app/src/main/res/values/strings.xml]: be sure to clean up TODO's

**Review by abunur — CHANGES_REQUESTED** [review]

File: Week04/MoviesApp/app/src/main/res/values/strings.xml (hunk 5 above)
Comment (daniel-waiguru) [reply Week04/MoviesApp/app/src/main/res/values/strings.xml]: Thank you
I will fix the above

**Pull request merged**